ui-library-add-call-recording/api/onRecordingFileStatusUpdated.cs

```diff
@@ -0,0 +1,142 @@
+            var payload = JsonConvert.DeserializeObject<RecordingFileStatusUpdatedPayload>(e.Data.ToString(), new JsonSerializerSettings { NullValueHandling = NullValueHandling.Ignore });
+
+            var callingServerClient = new CallingServerClient(Settings.GetACSConnectionString());
+            var storageClient = new BlobStorage(log, Settings.GetRecordingStoreConnectionString(), Settings.GetRecordingStoreContainerName());
+
+            var chunks = payload.RecordingStorageInfo.RecordingChunks;
+            var blobStorePaths = new List<string>();
+            foreach (var chunk in chunks)
+            {
+                var name = blobName(serverCallId, chunk);
+                var inStream = callingServerClient.DownloadStreaming(new Uri(chunk.ContentLocation)).Value;
+                if (await storageClient.UploadFileAsync(name, inStream))
+                {
+                    blobStorePaths.Add(name);
+                    log.LogInformation($"Transferred {name}");
+                }
+            }
+            log.LogInformation($"Uploaded {blobStorePaths.Count} chunks for {serverCallId}");
+        }
+
+        static ActionResult ValidationMessageResponse(EventGridEvent e)
+        {
+            var data = e.Data as JObject;
+            var eventData = data.ToObject<SubscriptionValidationEventData>();
+            var responseData = new SubscriptionValidationResponse
+            {
+                ValidationResponse = eventData.ValidationCode
+            };
+            if (responseData.ValidationResponse != null)
+            {
+                return new OkObjectResult(responseData);
+            }
+            return new OkResult();
+        }
+
+        static bool IsValidationMessage(EventGridEvent e)
+        {
+            return string.Equals(e.EventType, EventTypes.EventGridSubscriptionValidationEvent, StringComparison.OrdinalIgnoreCase);
+        }
+
+        static bool IsRecordingFileStatusUpdatedMessage(EventGridEvent e)
+        {
+            return string.Equals(e.EventType, "Microsoft.Communication.RecordingFileStatusUpdated", StringComparison.OrdinalIgnoreCase);
+        }
+
+        static string ExtractServerCallIDOrDie(string subject)
+        {
+            var match = serverCallIdExtractionPattern.Match(subject);
+            var serverCallId = match.Groups["serverCallId"].Value;
+            if (string.IsNullOrEmpty(serverCallId))
+            {
+                throw new Exception($"Failed parse serverCallId from {subject}");
+            }
+            return serverCallId;
+        }
+
+        static string blobName(string serverCallId, RecordingChunk chunk)
+        {
+            return $"call_{serverCallId}/chunk_{chunk.DocumentId}_{chunk.Index}.mov";
+        }
+
+        static Regex serverCallIdExtractionPattern = new Regex(@".*serverCallId/(?<serverCallId>[^/]*)/recordingId/.*");
+    }
+
+    class RecordingFileStatusUpdatedPayload

```

Comment:
My preference is to have classes in their own files. Its easier to read and refactor if we need to change anything. Happy to get push back here.

Reply:
Sure, I actually wanted to move this into a namespace as well, but will do that later.

Reply:
Moved to namespace. Want to keep `api/` folder clean so that you just see each Function handler there.